### PR TITLE
Complete ingress controller separation

### PR DESCRIPTION
**Pull request by kyessenov**
Split TPR controller into TPR and ingress stores using the aggregate controller.

### CODE REVIEW

**Comment by kyessenov** [comment]
you have to trust me that presubmit tests passed locally on my machine ;)

File: adapter/config/ingress/conversion.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+		err = multierror.Append(
+			fmt.Errorf("could not decode string into ingress rule name: %s", name),
+			pathErr, ruleErr)
+		return
+	}
+
+	return
+}
+
+// isRegularExpression determines whether the given string s is a non-trivial regular expression,
```
Comment (ijsnellf) [inline adapter/config/ingress/conversion.go]: Warn that Golang regular expression might not be a 1-1 match with this regular expression?

File: adapter/config/ingress/controller.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+	}
+
+	rules := convertIngress(*ingress, c.domainSuffix)
+	rule, exists := rules[key]
+	return rule, exists, ingress.GetResourceVersion()
+}
+
+func (c *controller) List(typ string) ([]model.Config, error) {
+	if typ != model.IngressRule {
+		return nil, nil
```
Comment (ijsnellf) [inline adapter/config/ingress/controller.go]: error?

File: adapter/config/ingress/controller_test.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+	if len(rules) != expectedRuleCount {
+		t.Errorf("expected %d IngressRule objects to be created, found %d", expectedRuleCount, len(rules))
+	}
+
+	for _, listMsg := range rules {
+		getMsg, exists, _ := ctl.Get(model.IngressRule, listMsg.Key)
+		if !exists {
+			t.Errorf("expected IngressRule with key %v to exist", listMsg.Key)
+
+			listRule := listMsg.Content.(*proxyconfig.IngressRule)
```
Comment (ijsnellf) [inline adapter/config/ingress/controller_test.go]: avoid panic.
```go
listRule, _ := listMsg.Content.(*proxyconfig.IngressRule)
getRule, _ := getMsg.(*proxyconfig.IngressRule)
```
or `listRule, ok := ...` and check `ok`

File: adapter/config/ingress/conversion.go (hunk 1 above)
Comment (kyessenov) [inline adapter/config/ingress/conversion.go]: added a comment. this function feels like a hack. I just moved it over.

File: adapter/config/ingress/controller.go (hunk 2 above)
Comment (kyessenov) [inline adapter/config/ingress/controller.go]: done

File: adapter/config/ingress/controller_test.go (hunk 3 above)
Comment (kyessenov) [inline adapter/config/ingress/controller_test.go]: done

**Pull request merged**